idpi/src/operators/vertical_interpolation.py

```diff
@@ -0,0 +1,363 @@
+"""Vertical interpolation operators."""
+
+import numpy as np
+import xarray as xr
+
+
+def interpolate_k2p(field, mode, pfield, tcp_values, tcp_units):
+    """Interpolate a field from model (k) levels to pressure coordinates.
+
+    Parameters
+    ----------
+    field : xarray.DataArray
+        field to interpolate
+    mode : str
+        interpolation algorithm, one of {"linear_in_tcf", "linear_in_lntcf", "nearest_sfc"}
+    pfield : xarray.DataArray
+        pressure field on k levels in Pa
+    tcp_values : list of float
+        target coordinate values
+    tcps_units : str
+        target coordinate units
+
+    Returns
+    -------
+    ftc : xarray.DataArray
+        field on target coordinates
+
+    """
+    # TODO: check missing value consistency with GRIB2 (currently comparisons are done with np.nan)
+    #       check that pfield is the pressure field, given in Pa
+    #       check that field and pfield are compatible
+    #       print warn message if result contains missing values
+
+    # Initializations
+    # ... supported interpolation modes
+    interpolation_modes = ("linear_in_tcf", "linear_in_lntcf", "nearest_sfc")
+    if mode not in interpolation_modes:
+        raise RuntimeError("interpolate_k2p: unknown mode", mode)
+    # ... supported tc units and corresponding conversion factors to Pa
+    tcp_unit_conversions = dict(Pa=1.0, hPa=100.0)
+    if tcp_units not in tcp_unit_conversions.keys():
+        raise RuntimeError("interpolate_k2p: unsupported value of tcp_units", tcp_units)
+    # ... supported range of tc values (in Pa)
+    tc_min = 1.0
```

Comment:
perhaps I find the naming tc_xxx a bit confusing, since it seems temperature. Perhaps pressure_min, pressure_values, etc would be more readable ?

Reply:
I will review the naming concept. I initially wanted to merge the routines back into one general interpolation routine, as we have it in fieldextra.

Reply:
Decided to see the interpolation routines as examples of different types of interpolations, and therefore explicitly used the names of the specific target coordinate field in the routines. Generalizations can be done when implementing the final application.